cosmos/ingestion/ingest/ingest.py

```diff
@@ -255,57 +263,144 @@ def enrich(self, file_path: str, dataset_id: str, pp_threshold: float, d_thresho
+        :param docs_and_tables_dfs: a tuple of a single pdf's COSMOS output consisting of all the rows for that pdf from
+        the non-aggregated parquet and the tables parquet, each as a pd.DataFrame
+        :return tables_df, missing_refs, statistics_df: input table dataframe with any enriched rows added,
+        df of pdf_name with any tables missed, df with pdf name and table detect recall, precision, and f1
+        :return type: Tuple(pd.DataFrame, pd.DataFrame, pd.DataFrame)
         """
 
         # disable SettingWithCopyWarning - safe since always over writing original
         pd.options.mode.chained_assignment = None
+        label_length = 2
+        context_column = 'context_from_text'
+        missing_refs = None
+        statistics_df = None
 
         # aggregate all doc words into one list of words, excepting tables and table captions
         doc_df = doc_and_tables_dfs[0]
         pp_classes_to_ignore = ['Table', 'Table Caption']
         all_doc_words = []
         for index, row in doc_df.iterrows():
             if row['postprocess_cls'] not in pp_classes_to_ignore:
-                all_doc_words.append(row['content'].split())
-
-        context_column = 'context_from_text'
+                all_doc_words.extend(row['content'].split())
 
+        # get table rows df and iteration copy, set pdf_name for error reporting
         tables_df = doc_and_tables_dfs[1]
-        tables_df[context_column] = None  # create enriched content column
+        tables_df[context_column] = None  # create empty enriched content column
         original_tables_df = tables_df.copy()  # copy to iterate over, copy to update
-        tables_df_columns = tables_df.columns.tolist()  # columns for writing final df
-        logger.debug(f'tables_df_columns:{tables_df_columns}')
+        tables_df_columns = tables_df.columns.tolist()  # columns list for outputting final df
         pdf_name = original_tables_df.pdf_name.to_list()[0]
 
-        label_length = 2
+        # join hyphenations in all words:
+        dehyphenated_words = []
+        i = 0
+        try:
+            while i < len(all_doc_words):
+                if all_doc_words[i][-1] == '-':
+                    if i + 1 <= len(all_doc_words):
+                        combine = (all_doc_words[i] + all_doc_words[i + 1]).replace("-", '')
+                        dehyphenated_words.append(combine)
+                        i += 2
+                    else:
+                        dehyphenated_words.append(all_doc_words[i])
+                        i += 1
+                else:
+                    dehyphenated_words.append(all_doc_words[i])
+                    i += 1
+        except (IndexError):
+            logger.info(f'DE-HYPHENATION FAILED ON {pdf_name} at index {i}: {all_doc_words[i]}')
+
+        all_doc_words = dehyphenated_words
+        label_match_pattern = 'table' + r'( ){1,2}[0-9.ivxlcdm-]+[\w]?'
```

Comment:
What's the justification for including all of these? How messy are the labels? Where there lots of things like "Table  -6" that loosening the regex catches?

Reply:
per step:
| component|justification|
|--|--|
|table| all table labels must start with 'table'*|
|( ){1,2}|there may be one or two space between 'table' and any index value|
|[0-9.ivxlcdm-]+|table index values maybe any Arabic or Roman numeral, and have a sub index separated by . or -. No length limit is specified here, less than 10 might be sensible. I have avoided introducing that assumption for now|
|[\w]?| optionally the index may have a single letter suffix e.g Table 3.4a|

*Now I'm thinking this actually should be r't( ){0,2}a( ){0,2}b( ){0,2}l( ){0,2}e' to handle times when spaces are present.

Its true that this might be looser than strictly necessary but I had intended err on the side of capturing more when it seemed like many 'tables' weren't being enriched. It's true that it now seems more like a precision issue than a recall issue though.